### ISSUE TITLE
[Discussion] Event listener to gather query stats/information

### ISSUE THREAD

**Issue by dharakk** [apache/pinot#7391]
Hi, We have been looking into doing query analysis in order to pin point a specific scenario and combing through server and broker logs is a difficult task. An event listener interface which enable us to add a hook into query execution and gather query info/stats would be really helpful. For example Presto provides the following interface to hook into query execution. https://github.com/prestodb/presto/blob/master/presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/EventListener.java 

Do we already have a similar mechanism in Pinot? Have we ever considered doing so as I can imagine it wont be as simple as Presto because of the query volume and latency sensitive nature of Pinot?

**Comment by mcvsubbu** [comment]
Yes, see https://github.com/apache/pinot/blob/master/pinot-broker/src/main/java/org/apache/pinot/broker/api/RequestStatistics.java

You will need to plug in your own mechanism to emit this from the broker

**Comment by kishoreg** [comment]
Good idea. As Subbu mentioned, we have all the information but we lack the listener interface. It's definitely worth adding one in our spi.

**Comment by ankitsultana** [comment]
Just saw this today. This was closed by #10606.
